lib/ipam/ipam.go

```diff
@@ -195,9 +195,54 @@ func (c ipamClient) getBlockFromAffinity(ctx context.Context, aff *model.KVPair)
 	return b, nil
 }
 
-func (c ipamClient) autoAssign(ctx context.Context, num int, handleID *string, attrs map[string]string, pools []net.IPNet, version ipVersion, host string) ([]net.IP, error) {
+// determinePools compares a list of requested pools with the enabled pools
+// and returns the intersect. If any requested pool does not exist, or is not enabled, an error is returned.
+// If no pools are requested, all enabled pools are returned.
+func (c ipamClient) determinePools(requestedPools []net.IPNet, version ipVersion) (pools []net.IPNet, err error) {
+	enabledPools, err := c.pools.GetEnabledPools(version.Number)
+	if err != nil {
+		log.WithError(err).Errorf("Error reading configured pools")
+		return nil, err
+	}
+	log.Debugf("enabled IPPools: %v", enabledPools)
+
+	if len(requestedPools) > 0 {
+		log.Debugf("requested IPPools: %v", requestedPools)
+		// Build a map so we can lookup existing pools.
+		pm := map[string]bool{}
+		for _, p := range enabledPools {
+			pm[p.String()] = true
+		}
+
+		// Make sure each requested pool exists.
+		for _, rp := range requestedPools {
+			if _, ok := pm[rp.String()]; !ok {
+				// The requested pool doesn't exist.
+				return nil, fmt.Errorf("the given pool (%s) does not exist, or is not enabled", rp.IPNet.String())
+			} else {
```

Comment:
nit: don't need the `else` here
(not reviewing but I was flying by and saw this)